src/modules/esl-animate/core/esl-animate-service.ts

```diff
@@ -0,0 +1,151 @@
+  groupDelay: number;
+
+  /** Do not unsubscribe after animate and repeat animation on each viewport intersection */
+  repeat?: boolean;
+}
+
+/** ESLAnimateService animation inner options. Contains system animation properties */
+interface ESLAnimateConfigInner extends ESLAnimateConfig {
+  /** @private animation requested */
+  _timeout?: number;
+  /** @private marker to unobserve */
+  _unsubscribe?: boolean;
+}
+
+/** Service to animate elements on viewport intersection */
+@ExportNs('AnimateService')
+export class ESLAnimateService {
+
+  /** ESLAnimateService default animation configuration */
+  protected static DEFAULT_CONFIG: ESLAnimateConfig = {cls: 'in', groupDelay: 100};
+  /** ESLAnimationService IntersectionObserver properties */
+  protected static OPTIONS_OBSERVER: IntersectionObserverInit = {threshold: [0.001, 0.2, 0.4, 0.6]};
+
+  /**
+   * Subscribe ESlAnimateService on element(s) to animate it on viewport intersection
+   * @param target - element(s) or elements to observe and animate
+   * @param config - optional animation configuration
+   */
+  public static observe(target: Element | Element[], config: string | Partial<ESLAnimateConfig> = {}): void {
+    wrap(target).forEach((item: Element) => this.instance.observe(item, config));
+  }
+
+  /** Unobserve element or elements */
+  public static unobserve(target: Element | Element[]): void {
+    wrap(target).forEach((item: Element) => this.instance.unobserve(item));
+  }
+
+  /** @returns if service observing target */
+  public static isObserved(target: Element): boolean {
+    return !!this.instance.getConfigFor(target);
+  }
+
+  @memoize()
+  private static get instance() {
+    return new ESLAnimateService();
+  }
+
+  protected _io = new IntersectionObserver(this.onIntersect, ESLAnimateService.OPTIONS_OBSERVER);
+  protected _entries: Set<Element> = new Set();
+  protected _configMap = new WeakMap<Element, ESLAnimateConfigInner>();
+
+  protected deferredOnAnimate = debounce(() => this.onAnimate(), 100);
+
+  /**
+   * Subscribe ESlAnimateService on element(s) to animate it on viewport intersection
+   * @param el - element or elements to observe and animate
+   * @param config - optional animation configuration
+   */
+  public observe(el: Element, config: string | Partial<ESLAnimateConfig> = {}): void {
+    const cfg = this.setConfigFor(el, config);
+    cfg.force && CSSClassUtils.remove(el, cfg.cls);
+    this._io.observe(el);
+  }
+
+  /** Unobserve element or elements */
+  public unobserve(el: Element): void {
+    this._io.unobserve(el);
+    this._configMap.delete(el);
+  }
+
+  /** Intersection observable callback */
+  @bind
+  protected onIntersect(entries: IntersectionObserverEntry[]): void {
+    entries.forEach(({target, intersectionRatio, isIntersecting}: IntersectionObserverEntry) => {
+      const config = this.getConfigFor(target);
+      if (!config) return;
+
+      if (intersectionRatio >= 0.4) this._entries.add(target);
```

Comment:
Perhaps it makes sense to define values via class properties? For ease of inheritance and overriding.

Reply:
Updated, thanks for the suggestion 👍🏻